File: 16Jan2024.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Random;

class RandomizedSet {
    HashMap<Integer, Integer> map;
    List<Integer> list;
    Random random;

    public RandomizedSet() {
        map = new HashMap<>();
        list = new ArrayList<>();
        random = new Random();
    }

    public boolean insert(int val) {
        if (map.containsKey(val))
            return false;
        map.put(val, list.size());
        list.add(val);
        return true;
    }

    public boolean remove(int val) {
        if (!map.containsKey(val))
            return false;
        int idx = map.get(val);
        list.set(idx, list.get(list.size() - 1));
        map.put(list.get(idx), idx);
        map.remove(val);
        list.remove(list.size() - 1);
        return true;
    }

    public int getRandom() {
        int idx = random.nextInt(list.size());
        return list.get(idx);
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */

File: 16Jan2024.py
import random


class RandomizedSet:
    
    def __init__(self):
        self.list = []
        self.map = {}
    
    def search(self, val:int) -> bool :
        return val in self.map
    
    def insert(self, val: int) -> bool:
        if(self.search(val)) :
            return False
        self.map[val]=len(self.list)
        self.list.append(val)
        return True

    def remove(self, val: int) -> bool:
        if(not self.search(val)) :
            return False
        idx = self.map[val]
        self.list[idx] = self.list[-1]
        
        self.map[self.list[idx]]=idx
        del self.map[val]
        self.list.pop()
        return True

    def getRandom(self) -> int:
        return random.choice(self.list)


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()